File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.java
class Solution {
    public int[] rearrangeArray(int[] nums) {
        int i=0,j=1,k=0;
        int n = nums.length;
        int[] res = new int[n];
        while(k<n){
            if(nums[k] >= 0){
                res[i] = nums[k];
                i +=2;
            }
            else{
                res[j]= nums[k];
                j += 2;
            }
            k++;
        }

        return res;
 
    }
}

File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        vector<int>res;
        int cnt = 0;
        int i=0;
        int j=0;
        int n = nums.size();
         while(i<n && j<n){
           while(nums[i]<0) i++;
           while(nums[j]>0) j++;
 
        if(cnt == 0 && nums[i] > 0){
            res.push_back(nums[i]);
            cnt=1;
        } 
        if(cnt == 1 && nums[j] < 0){
            res.push_back(nums[j]);
            cnt =0;
        }

        i++;
        j++;
        }
        return  res;
    }
};